src/dungeon_ai.c

```diff
@@ -43,6 +43,80 @@ extern void sub_807AB38(Entity *, u32);
 extern void sub_8041888(u32);
 extern u8 sub_803F428(s16 *);
 extern void sub_803E708(u32, u32);
+void sub_804AE84(Position *pos);
+extern void sub_8049ED4();
+extern void sub_8073D14(Entity *);
+extern void sub_807FE9C(Entity *, Position *, u32, u32);
+
+void nullsub_97(Entity *entity)
+{}
+
+void sub_8075708(Entity *entity)
```

Comment:
This doesn't seem like AI code anymore, seeing that the Trap Seer logic is in this function. Might be better to move this code to a different file.